### PR TITLE
Pull makeImage into its own package (from screenshoter)

### PR DESCRIPTION
**Pull request by tourdedave**
- [x] pull it into its own package
- [x] convert it to TS (e.g., no build errors and tests passing)
- [x] specify concrete types (not just `any`)
- [x] update screenshoter to use the new package
- [x] publish new package and update screenshoter package.json to use it (once code review is complete)

### CODE REVIEW

File: js/packages/image/src/index.ts [hunk 1]
```diff
@@ -664,4 +693,4 @@ function _scaleImageIncrementally(src, dst) {
   return dst
 }
 
-module.exports = makeImage
+export = makeImage
```
Comment (kyryloonufriiev) [inline js/packages/image/src/index.ts:696]: It would be better to use named export.

File: js/packages/image/src/index.ts [hunk 2]
```diff
@@ -1,13 +1,42 @@
...
+
+type TransformData = {
+  rotate: number,
+  scale: number,
+  crop?: any,
+  modifiers: any[],
+  added?: any,
+}
+
+type Image = {
```
Comment (kyryloonufriiev) [inline js/packages/image/src/index.ts:16]: I would specify it as an `interface`, not `type`. There is no big difference, but I usually use `type` to specify the shape of a data object, and `interfaces` to specify a functional object as we have here.

**Review by kyryloonufriiev — CHANGES_REQUESTED** [review]
Looks good to me. One small thing that is maybe more about esthetic is I wouldn't put the whole implementation in the index file, but rather create an `image.ts` file with the implementation of the image logic. And the same for test `index.d.ts` is not informative enough as for me.

File: js/packages/image/src/image.ts [hunk 3]
```diff
@@ -1,13 +1,42 @@
...
+
+type TransformData = {
+  rotate: number
+  scale: number
+  crop?: any
+  modifiers: any[]
+  added?: any
+}
+
+interface Image {
```
Comment (kyryloonufriiev) [inline js/packages/image/src/image.ts:16]: This should be exported, so we can use the time in other packages
```suggestion
export interface Image {
```

**Review by kyryloonufriiev — CHANGES_REQUESTED** [review]
Looks good to me, the only missing point is a dedicated github action to publish this pacakge.

**Comment by tourdedave** [comment]
> Looks good to me, there are a few missing points:
> 
> * Dedicated github action to publish `image` package.
> * Add `@applitools/scripts` to the `screenshoter` as a dev dependency.
> * Add `test:sanity` command to the `image`, it could run all of the tests, since they are fast.

`@applitools/scripts` is already a dev dep in `screenshoter`. The other items are done.

File: js/packages/image/package.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    "directory": "js/packages/image"
+  },
+  "license": "SEE LICENSE IN LICENSE",
+  "author": {
+    "name": "Applitools Team",
+    "email": "team@applitools.com"
+  },
+  "aliases": [
+    "screenshoter"
+  ],
```
Comment (kyryloonufriiev) [inline js/packages/image/package.json:21]: Aha. I see what confused me. We use the first alias as the name of the job, so it says that `screenshoter` doesn't have `scripts` dependency when the `image` is the one that doesn't have it. 
```suggestion
  "aliases": [
    "image"
  ],
```

**Review by kyryloonufriiev — CHANGES_REQUESTED** [review]
If after merging my proposed change all tests are passing in the PR, feel free to merge.

**Review by kyryloonufriiev — APPROVED** [review]
Great, let's merge and release it!

**Pull request merged**